sentry-android-core/src/main/java/io/sentry/android/core/ScreenshotEventProcessor.java

```diff
@@ -51,10 +54,86 @@ public ScreenshotEventProcessor(
     application.registerActivityLifecycleCallbacks(this);
   }
 
+  public static ScreenshotEventProcessor createInstance(
+      final @NotNull Application application,
+      final @NotNull SentryAndroidOptions options,
+      final @NotNull BuildInfoProvider buildInfoProvider) {
+    ScreenshotEventProcessor.instance =
+        new ScreenshotEventProcessor(application, options, buildInfoProvider);
+    return ScreenshotEventProcessor.instance;
+  }
+
+  public static ScreenshotEventProcessor getInstance() {
+    return ScreenshotEventProcessor.instance;
+  }
+
+  public void setCurrentActivity(@NonNull Activity activity) {
+    if (currentActivity != null && currentActivity.get() == activity) {
+      return;
+    }
+    currentActivity = new WeakReference<>(activity);
+  }
+
+  public byte[] takeScreenshot() {
+    if (currentActivity == null) {
+      return null;
+    }
+
+    final Activity activity = currentActivity.get();
+    if (!isActivityValid(activity)
+        || activity.getWindow() == null
+        || activity.getWindow().getDecorView() == null
+        || activity.getWindow().getDecorView().getRootView() == null) {
+      this.options
+          .getLogger()
+          .log(SentryLevel.DEBUG, "Activity isn't valid, not taking screenshot.");
+      return null;
+    }
+
+    final View view = activity.getWindow().getDecorView().getRootView();
+    if (view.getWidth() <= 0 || view.getHeight() <= 0) {
+      this.options
+          .getLogger()
+          .log(SentryLevel.DEBUG, "View's width and height is zeroed, not taking screenshot.");
+      return null;
+    }
+
+    try {
+      // ARGB_8888 -> This configuration is very flexible and offers the best quality
+      final Bitmap bitmap =
+          Bitmap.createBitmap(view.getWidth(), view.getHeight(), Bitmap.Config.ARGB_8888);
+
+      final Canvas canvas = new Canvas(bitmap);
+      view.draw(canvas);
+
+      final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+
+      // 0 meaning compress for small size, 100 meaning compress for max quality.
+      // Some formats, like PNG which is lossless, will ignore the quality setting.
+      bitmap.compress(Bitmap.CompressFormat.PNG, 0, byteArrayOutputStream);
+
+      if (byteArrayOutputStream.size() <= 0) {
+        this.options
+            .getLogger()
+            .log(SentryLevel.DEBUG, "Screenshot is 0 bytes, not attaching the image.");
+        return null;
+      }
+
+      // screenshot png is around ~100-150 kb
+      return byteArrayOutputStream.toByteArray();
+    } catch (Throwable e) {
+      this.options.getLogger().log(SentryLevel.ERROR, "Taking screenshot failed.", e);
+    }
+    return null;
+  }
+
   @SuppressWarnings("NullAway")
   @Override
   public @NotNull SentryEvent process(final @NotNull SentryEvent event, @NotNull Hint hint) {
-    if (!lifecycleCallbackInstalled) {
+    if (HintUtils.getIsFromHybridSdk(hint)
```

Comment:
M: I'd move the `getIsFromHybridSdk` later otherwise Native events pay the price every time even if the integration isn't installed nor its an error event.